src/extension.ts

```diff
@@ -103,3 +152,110 @@ export async function activateInternal(context: vscode.ExtensionContext, perfSta
+                    await requestUtils.downloadFunctionAppContent(defaultHostName, downloadFilePath, listKeyResponse.parsedBody.masterKey);
+                });
+            }
+
+            const projectFilePathUri: vscode.Uri = vscode.Uri.joinPath(vsCodeFilePathUri, `${functionAppName}`);
+            const projectFilePath: string = projectFilePathUri.fsPath;
+            const devContainerFolderPathUri: vscode.Uri = vscode.Uri.joinPath(projectFilePathUri, '.devcontainer');
+            ext.context.globalState.update(LocalDevelopmentGlobalStates.projectFilePath, projectFilePathUri.fsPath);
+            await callWithTelemetryAndErrorHandling('azureFunctions.extractContentAndDownloadDevContainerFiles', async (_actionContext: IActionContext) => {
+                // tslint:disable-next-line: no-unsafe-any
+                await extract(downloadFilePath, { dir: projectFilePath });
+                await requestUtils.downloadFile(
+                    `https://raw.githubusercontent.com/microsoft/vscode-dev-containers/master/containers/${devContainerName}/.devcontainer/devcontainer.json`,
+                    vscode.Uri.joinPath(devContainerFolderPathUri, 'devcontainer.json').fsPath
+                );
+                await requestUtils.downloadFile(
+                    `https://raw.githubusercontent.com/microsoft/vscode-dev-containers/master/containers/${devContainerName}/.devcontainer/Dockerfile`,
+                    vscode.Uri.joinPath(devContainerFolderPathUri, 'Dockerfile').fsPath
+                );
+            });
+            vscode.window.showInformationMessage(localize('restartingVsCodeInfoMessage', 'Restarting VS Code with your function app project'));
+            vscode.commands.executeCommand('vscode.openFolder', vscode.Uri.file(projectFilePath));
+        } catch (err) {
+            vscode.window.showErrorMessage(localize('failedLocalProjSetupErrorMessage', 'Failed to set up your local project. Please try again.'));
+        } finally {
+            vscode.workspace.fs.delete(
+                vscode.Uri.file(toBeDeletedFolderPathUri.fsPath),
+                {
+                    recursive: true,
+                    useTrash: true
+                }
+            );
+        }
+    } else {
+        vscode.window.showErrorMessage(localize('invalidInputsErrorMessage', 'Invalid inputs. Please try again.'));
+    }
+}
+
+function getProjectLanguageForLanguage(language: string): ProjectLanguage {
+    switch (language) {
+        case 'powershell':
+            return ProjectLanguage.PowerShell;
+        case 'node':
+            return ProjectLanguage.TypeScript;
+        case 'python':
+            return ProjectLanguage.Python;
+        case 'java8':
+        case 'java11':
+            return ProjectLanguage.Java;
+        case 'dotnetcore2.1':
+        case 'dotnetcore3.1':
+            return ProjectLanguage.CSharp;
+        default:
+            throw new Error(`Language not supported: ${language}`);
```

Comment:
should we localize this error message? it might be a good idea to show to the user if this is opened for an unsupported language. E.g. .NET 5

Reply:
Sure, I can add a info error for user which will be localized and then throw